### ISSUE TITLE
Adding filters for views

### ISSUE THREAD

**Issue by s3696369** [XRM-OSS/D365-PowerKanban#31]
Firstly, this is an amazing extension for model driven apps. It has been really well received here. We all love it! Thank you so much.

I am using it on a view of tasks, and I wondered if there's a way to add in filters? What I am trying to do is to be able to filter it by the owner, but there may be other columns that could be used an filters too. At the moment we have created different views to show each person's tasks, but it's not ideal if there's a lot more people.

The other question I had is about opening the record. When you click the arrow, it opens on a new page, is there a way to change this to open in modal instead? This is the preferred method here. I know that we can do that by using the down arrow options, this would be a neat enhancement for us.

**Comment by DigitalFlow** [comment]
Hi  @s3696369,

first up, I'm really glad that you like it. It's great to receive some feedback.
Which version do you currently have installed? The latest one currently is v0.9.15: https://github.com/XRM-OSS/D365-PowerKanban/releases/tag/v0.9.15

In this version there is still no filter, but you can type the user name in the search box and it will search for the name in all fields that can be found in the data.
Adding a filter is a good idea, I will create an issue for the extension.

As to opening the records: 
In the latest version, clicking the arrow should open the record in the same tab. When clicking the small arrow down and choosing "Open In Modal" it will open the record in a modal, just as you said.
I could probably add a config option on how to open the record by the default arrow, I will create an issue for this extension as well.

Kind Regards,
Florian

**Comment by s3696369** [comment]
Hi Florian,

Thanks for your reply!

I am using the current version. The search does the trick, but there are a couple of tasks where the description references another person who isn't assigned the task if that makes sense?

Being able to change the way the record opens by default would be awesome! It is still fantastic the way it is now, a huge improvement over using a standard view.

Thank you,
Zuleika

**Comment by DigitalFlow** [comment]
Hi Zuleika,

no problem, you're welcome.

One thought about your current situation:
You said you created views for filtering tasks for the time being. This is what I recommend as well right now, but you said that it might become not ideal if you have more people.
This sounds as if you created one view for each person that you want to filter records for? If this is what you did, then there is a better way: Inside the view you can set the filter for the user to "Equals current user".
This way you only have one view for "My tasks" instead of one per user.

Kind Regards,
Florian

**Comment by s3696369** [comment]
Hi Florian,

I have a view where the user = current user, but the reason for having one for each person/group of people would be a situation where a manager wants to check tasks for their team. Luckily we are fairly small at the moment so it hasn't been a problem creating individual views for this purpose. In this situation, being able to filter all the tasks by selected people would be amazing. I'm sure this is possible if you set up roles or something, but we have a fairly flat structure due to our size, so while I am part of a dev team, I might need to see what tasks our marketing person has.

I'm not sure if that makes sense.

I really appreciated that you have taken the time to assist.

**Comment by DigitalFlow** [comment]
Hi Zuleika,

ah yes that makes sense, I did not think of the scenario of managers wanting to filter for their team's data.
I created #32 and #33 for implementing the use cases you described, #32 should be implemented fairly soon.

Kind Regards,
Florian

**Comment by DigitalFlow** [comment]
Hi Zuleika,

Issue #32 was just implemented and released as v0.9.16.
After upgrading, you can set a property `defaultOpenHandler` on the primaryEntity or secondaryEntity config property to `modal` for making the arrow button open in modal by default.

It could look something like this:
```JSON
{
    "primaryEntity": {
        "logicalName": "incident",
        "swimLaneSource": "statuscode"
        "defaultOpenHandler": "modal"
    },
    "secondaryEntity": {
        "logicalName": "task",
        "parentLookup": "regardingobjectid",
        "swimLaneSource": "statuscode",
        "defaultOpenHandler": "modal"
    }
}
```
Please test, some feedback would be appreciated.
As for your other request regarding filtering: I created #34 for keeping track of this requirement and would like to close this issue in the meantime. Are you ok with that?

Kind Regards,
Florian

**Comment by s3696369** [comment]
Hi Florian,

It is absolutely wonderful! I am extremely grateful for this change. You can close this one off, it looks to be working perfectly.

Thank you so much.
